### ISSUE TITLE
The stories section sometimes do not appear if recommendations are turned off.

### ISSUE THREAD

**Issue by gerwld** [gerwld/igplus-extension#4]
## The stories section sometimes do not appear if recommendations are turned off.

**- OS:** _any / not specified_
**- Browser:** _any (?), probably Firefox_

**Comment by devacius** [comment]
Hello Gerwld, got to know about this amazing project . I want to work on this issue . So, can you give this issue to me also, please tell me the installation and setup projects.(Note- already followed the ones in src/HOW_TO_FOR_MOD but my browser load unpack doesn't seem to get the backgroud.js file and no public folder is being build).

**Comment by jc97** [comment]
I've discovered this issue too. However, I've found out, it seems to be independent from the extension. When I completely disable IGPlus and then directly open the "following" page of Instagram, the stories are not loaded neither. The same happens when I open the start page of Instagram and immediately (before the stories are loaded on the main page) press the link for the "following" page of Instagram.

I am using Firefox on Android. With Firefox on Linux the issue doesn't occur so far.

**Comment by devacius** [comment]
Hello gerwld, I am using brave in windows and yes the same issues are there
in windows too. Would you like me to work on it?

**Comment by gerwld** [comment]
> please tell me the installation and setup projects.(Note- already followed the ones in src/HOW_TO_FOR_MOD but my browser load unpack doesn't seem to get the backgroud.js file and no public folder is being build).

Hello @devacius! You should have gulp installed, also it does not work in browsers without being built. If you're trying to work with it in dev mode, then build it first using the `pnpm run dev` script. This should create the public/chromium directory, and you have to select it in your browser (not the root directory). If it's not, give me some further information to help you with it.

**Comment by gerwld** [comment]
> I've discovered this issue too. However, I've found out, it seems to be independent from the extension. When I completely disable IGPlus and then directly open the "following" page of Instagram, the stories are not loaded neither. The same happens when I open the start page of Instagram and immediately (before the stories are loaded on the main page) press the link for the "following" page of Instagram.

@jc97 yes, I can assume it's a good observation. It seems like Instagram provides the stories section on the `/?variant=following` path screen only if data for stories was fetched before on the main page. 

Like, it makes common sense if it uses a scenario where the link for `/?variant=following` is only provided on the main page, so the user will fetch it anyway. But it is not as good for those trying to navigate it straightforwardly as IGPlus does. But I believe there is 100% some way to fix it.

**Comment by jc97** [comment]
Probably, an easy way to fix it, is navigating to the main page first and then automatically redirect to the "following" page.

**Comment by gerwld** [comment]
> Probably, an easy way to fix it, is navigating to the main page first and then automatically redirect to the "following" page.

Yes, it may be a good solution! I'll check if it looks good. Maybe some form of delay on the initial load may help, at least.

**Comment by jc97** [comment]
Are there any news on this issue?

**Comment by jc97** [comment]
I just tried the current version once again and the problem seems to be fixed!

**Comment by gerwld** [comment]
> I just tried the current version once again and the problem seems to be fixed!

No, I thought the same, but just recieved an email that it's still persist. There should be an option to trigger that fetching request out of "?variant=following" in a Mozilla browser. In Chrome (as I can see) it works fine.

**Comment by gerwld** [comment]
> Hello gerwld, I am using brave in windows and yes the same issues are there
> in windows too. Would you like me to work on it?
> […](#)

Yes, for sure you can try. I accept any beneficial pull requests, and opened to any ideas. There's definitely a fix for that issue, and many users are waiting for it. 

I will add a full Cotribution FAQ later, but for now the dev build can be accessed by `$ pnpm i && pnpm chrome` command. Most likely it will work with npm and yarn as well

**Comment by gerwld** [comment]
> If this issue still needs help I think I can give it a try. I did some preliminary testing and adding the rules
> 
> ```
> div ~ article {
>     display: None;
> }
> article ~ div {
>     display: None;
> }
> ```
> 
> is enough to hide the suggested posts, but it gets stuck in a loop trying to load new posts and hiding them. If I can find a way to kill those callbacks too then I should be able to do this pretty easily I think.
> 
> This method would let you use the regular home page with stories, use the other IGPlus features to remove reels and stuff, and removes the suggested posts that appear after seeing posts from those you follow.

The problem you're facing is the obfuscated callback under IIFE that is called each time when the Observer API figures out that latest fetched post div is at a certain point of a screen view. You cannot just disable it (sorry), as it will break the post fetching method. But, If you're proposing to disable the posts completely there's an option called "Disable the Feed" which implements that scenario in IGPlus
